test/ruby/test_keyword.rb

```diff
@@ -182,6 +182,13 @@ def test_method_parameters
                   [:keyrest, :kw], [:block, :b]], method(:f9).parameters)
   end
 
+  def test_keyword_with_anonymous_keyword_splat
+    def self.a(b: 1, **) [b, **] end
+    kw = {b: 2, c: 3}
+    assert_equal([2, {c: 3}], a(**kw))
+    assert_equal({b: 2, c: 3}, kw)
+  end
```

Comment:
3af47f14d4e seems to pass this.

Reply:
Yes.  However, if you commit the `compile.c` change without the `vm_args.c` change, and you merge #9776 to avoid the unnecessary array allocation, I think it fails.